### PR TITLE
[WEBDATA-120] Fix password confirmation

### PR DESCRIPTION
**Pull request by DanielBrookRoberge**
This PR fixes the bug preventing registration of new user accounts. It also reads the Magento messages cookie to get the correct error message for the register and login forms.

 **JIRA**: https://mobify.atlassian.net/browse/WEBDATA-120
 **Linked PRs**: (links to corresponding PRs, optional)

## Changes
- No longer pass the password confirmation value into the command
- Duplicate the password in the Merlin's connector for registration
- Check that the passwords match in the confirmation page registration form
- Read the error message out of the `mage-messages` cookie rather than using a hard-coded value

## How to test-drive this PR
- Preview Merlin's
- Attempt to register with a new email address (e.g. `yourname+webdata120@mobify.com`) 
- See that it works

### CODE REVIEW

File: web/app/containers/login/actions.js [hunk 1]
```diff
@@ -120,10 +120,9 @@ export const submitRegisterForm = (formValues) => (dispatch) => {
         firstname,
         lastname,
         email,
-        password,
-        password_confirmation
```
Comment (jeremywiebe) [inline web/app/containers/login/actions.js]: Should we be doing the same comparison between `password` and `password_confirmation` here as we do in the checkout register step?

Comment (DanielBrookRoberge) [inline web/app/containers/login/actions.js]: The `validateRegisterForm` function already does this, so I just left it alone.

File: web/app/integration-manager/_merlins-connector/account/commands.js [hunk 2]
```diff
@@ -30,25 +31,30 @@ export const initRegisterPage = (url) => (dispatch) => {
...
-                throw new SubmissionError(error)
+                const messages = JSON.parse(decodeURIComponent(getCookieValue(MAGENTO_MESSAGE_COOKIE)))
+
+                throw new SubmissionError({
+                    _error: (messages[0] || defaultErrorMessage).text.replace(/\+/g, ' ')
+                })
             }
             return '/customer/account'
         })
-        .catch((error) => {
```
Comment (jeremywiebe) [inline web/app/integration-manager/_merlins-connector/account/commands.js]: Nice!

File: web/app/containers/checkout-confirmation/actions.js [hunk 3]
```diff
@@ -40,7 +41,15 @@ export const submitRegisterForm = () => {
             billingAddressData
         } = registrationFormSelector(getState())
 
-        return dispatch(registerUser(firstname, lastname, email, password, password_confirmation))
+        /* eslint-disable camelcase */
+        if (password !== password_confirmation) {
```
Comment (jnurse) [inline web/app/containers/checkout-confirmation/actions.js]: This shouldn't be necessary if we've got a validate function checking the same for the form.

**Pull request merged**